### PR TITLE
AO3-4471 Cache the fact that there is not a site banner.

### PR DESCRIPTION
**Pull request by zz9pzza**
Make sure that the site is happy with or with out a banner. ( I believe this is low risk )

https://otwarchive.atlassian.net/browse/AO3-4471

### CODE REVIEW

**Comment by sarken** [comment]
Can you address Hound's comments, please? Thanks!

**Comment by zz9pzza** [comment]
@sarken done

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -114,7 +114,13 @@ def load_admin_banner
     if Rails.env.development?
       @admin_banner = AdminBanner.where(:active => true).last
     else
-      @admin_banner = Rails.cache.fetch("admin_banner"){AdminBanner.where(:active => true).last}
+      # http://stackoverflow.com/questions/12891790/will-returning-a-nil-value-from-a-block-passed-to-rails-cache-fetch-clear-it
+      # Basically we need to store a nil separately.
+      @admin_banner = Rails.cache.fetch("admin_banner") do 
+        banner=AdminBanner.where(:active => true).last
```
Comment (scottsds) [inline app/controllers/application_controller.rb]: There needs to be spaces on either sides of the `=`.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -114,7 +114,13 @@ def load_admin_banner
...
+        banner.nil? ? "" : banner
+      end
+      if @admin_banner == "" then @admin_banner = nil end
```
Comment (scottsds) [inline app/controllers/application_controller.rb]: Could this be something like `@admin_banner = nil if @admin_banner.blank?`

**Comment by scottsds** [comment]
When this passes Travis, it's good to go.

**Pull request merged**